satpy/composites/__init__.py

```diff
@@ -551,69 +551,79 @@ def _insert_palette_colors(channels, palette):
+                                "day_only" means only day portion will be kept
+                                "night_only" means only night portion will be kept
 
         """
         self.lim_low = lim_low
         self.lim_high = lim_high
+        self.day_night = day_night
         super(DayNightCompositor, self).__init__(name, **kwargs)
 
     def __call__(self, projectables, **kwargs):
         """Generate the composite."""
         projectables = self.match_data_arrays(projectables)
 
-        day_data = projectables[0]
-        night_data = projectables[1]
+        foreground_data = projectables[0]
+        background_data = projectables[1] if self.day_night == "day_night" else foreground_data
 
         lim_low = np.cos(np.deg2rad(self.lim_low))
         lim_high = np.cos(np.deg2rad(self.lim_high))
         try:
-            coszen = np.cos(np.deg2rad(projectables[2]))
+            coszen = np.cos(np.deg2rad(projectables[2 if self.day_night == "day_night" else 1]))
         except IndexError:
             from pyorbital.astronomy import cos_zen
             LOG.debug("Computing sun zenith angles.")
             # Get chunking that matches the data
             try:
-                chunks = day_data.sel(bands=day_data['bands'][0]).chunks
+                chunks = foreground_data.sel(bands=foreground_data['bands'][0]).chunks
             except KeyError:
-                chunks = day_data.chunks
-            lons, lats = day_data.attrs["area"].get_lonlats(chunks=chunks)
-            coszen = xr.DataArray(cos_zen(day_data.attrs["start_time"],
+                chunks = foreground_data.chunks
+            lons, lats = foreground_data.attrs["area"].get_lonlats(chunks=chunks)
+            coszen = xr.DataArray(cos_zen(foreground_data.attrs["start_time"],
                                           lons, lats),
                                   dims=['y', 'x'],
-                                  coords=[day_data['y'], day_data['x']])
+                                  coords=[foreground_data['y'], foreground_data['x']])
         # Calculate blending weights
         coszen -= np.min((lim_high, lim_low))
         coszen /= np.abs(lim_low - lim_high)
         coszen = coszen.clip(0, 1)
 
         # Apply enhancements to get images
-        day_data = enhance2dataset(day_data)
-        night_data = enhance2dataset(night_data)
+        foreground_data = enhance2dataset(foreground_data)
+        background_data = enhance2dataset(background_data)
 
         # Adjust bands so that they match
         # L/RGB -> RGB/RGB
         # LA/RGB -> RGBA/RGBA
         # RGB/RGBA -> RGBA/RGBA
-        day_data = add_bands(day_data, night_data['bands'])
-        night_data = add_bands(night_data, day_data['bands'])
+        foreground_data = add_bands(foreground_data, background_data['bands'])
+        background_data = add_bands(background_data, foreground_data['bands'])
```

Comment:
The alpha band is only needed for day_only or night_only, right? Otherwise we assume that the two composites will completely fill the image in the `day_night` case.

So could you do:

```python
background_bands = background_data['bands'] + ('A',) if background_data['bands'][-1] != 'A' else background_data['bands']
foreground_data = add_bands(foreground_data, background_bands)
```

Reply:
background_data['bands'] is not a list like ['R', 'G', 'B'] so I can't just simply `+ ('A')`. I need to first define that **"A"**. So I  made a part just from `add_bands` to ensure an alpha channel added. By this, we'll leave `add_bands` its test alone.